src/taker/api2.rs

```diff
@@ -0,0 +1,1376 @@
+
+            let first_maker_pubkey = first_maker.offer.tweakable_point.inner;
+            let outgoing_contract_my_pubkey = self.ongoing_swap_state.outgoing_contract_my_pubkey
+                .ok_or_else(|| TakerError::General("No taker pubkey found".to_string()))?;
+            let outgoing_contract_internal_key =
+                crate::protocol::musig_interface::get_aggregated_pubkey_i(
+                    outgoing_contract_my_pubkey.inner,
+                    first_maker_pubkey
+                );
+
+            // Create taproot script (P2TR output)
+            let (outgoing_contract_taproot_script, outgoing_contract_taproot_spendinfo) =
+                create_taproot_script(hashlock_script, timelock_script, outgoing_contract_internal_key);
+
+            self.ongoing_swap_state.outgoing_contract_internal_key =
+                Some(outgoing_contract_internal_key);
+            self.ongoing_swap_state.outgoing_contract_tap_tweak =
+                Some(outgoing_contract_taproot_spendinfo.tap_tweak().to_scalar());
+
+            let outgoing_contract_taproot_address =
+                bitcoin::Address::from_script(&outgoing_contract_taproot_script, bitcoin::Network::Regtest).map_err(
+                    |e| TakerError::General(format!("Failed to create taproot address: {:?}", e)),
+                )?;
+
+            let signed_outgoing_contract_tx = {
+                use crate::utill::DEFAULT_TX_FEE_RATE;
+                use crate::wallet::Destination;
+
+                let funding_utxo_info = self
+                    .wallet
+                    .get_utxo((funding_utxo.txid, funding_utxo.vout))?
+                    .ok_or_else(|| TakerError::General("Funding UTXO not found".to_string()))?;
+
+                // Use Destination::Multi to send exact amount and get change back
+                let signed_tx = self.wallet.spend_from_wallet(
+                    DEFAULT_TX_FEE_RATE,
+                    Destination::Multi {
+                        outputs: vec![(outgoing_contract_taproot_address, self.ongoing_swap_state.swap_params.send_amount)],
+                        op_return_data: None,
+                    },
+                    &[(funding_utxo.clone(), funding_utxo_info)],
+                )?;
+
+                signed_tx
+            };
+
+            contract_transactions.push(signed_outgoing_contract_tx);
+        } else {
+            return Err(TakerError::General("No makers chosen for swap".to_string()));
+        }
+
+        Ok(contract_transactions)
+    }
+
+    /// Wait for transactions to be confirmed on the blockchain
+    /// 
+    /// This function polls the wallet to check if the given transactions are confirmed.
+    /// It will retry for a maximum number of attempts with a delay between each attempt.
+    fn wait_for_transaction_confirmations(&self, transactions: &[Transaction]) -> Result<(), TakerError> {
+        use std::thread;
+        use std::time::Duration;
```

Comment:
Same for this. We already have a `watch_for_txs()` routine. We can use that instead?